pkg/util/filesystem/permission_nowindows.go

```diff
@@ -27,24 +30,38 @@ func NewPermission() (*Permission, error) {
 	return &Permission{}, nil
 }
 
-// RestrictAccessToUser sets the file user and group to the same as 'dd-agent' user. If the function fails to lookup
-// "dd-agent" user it return nil immediately.
-func (p *Permission) RestrictAccessToUser(path string) error {
+// return the user dd-agent uid and gid
+func UserDDAgent() (found bool, usrID int, grpID int, err error) {
```

Comment:
I don't think you need the `found` return value. If `err != nil` then it is assumed `usrID` and `grpID` are invalid.